core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -114,6 +114,10 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
   // to be defined in subclass to create a specific fetcher
   def createFetcherThread(fetcherId: Int, sourceBroker: BrokerEndPoint): T
 
+  def updateEpochs(partitionAndOffsets: Map[TopicPartition, InitialFetchState]): Unit = lock synchronized {
```

Comment:
Does this need to be synchronized here? Shouldn't it only be called from becomeLeaderOrFollower which holds the replicaStateChangeLock?

Reply:
The other methods used to update inner threads are in synchronized lock so this new method is synchronized also.